### PR TITLE
Added new command for pack dependencies  

### PR DESCRIPTION
**Pull request by IkaDemisto**
## Status
Ready

## Related Issues
related to: https://github.com/demisto/etc/issues/22129

## Description
Added new `demisto-sdk` command that uses id set and searches for specific pack dependencies. Writes back the found result to `pack_metadata.json`.

### CODE REVIEW

**Comment by IkaDemisto** [comment]
Id set file is for unit tests

File: demisto_sdk/__main__.py [hunk 1]
```diff
@@ -589,10 +591,27 @@ def id_set_command(**kwargs):
     id_set_creator.create_id_set()
 
 
+# ====================== find-dependencies ====================== #
+@main.command(name="find-dependencies",
```
Comment (bakatzir) [inline demisto_sdk/__main__.py]: add `hidden=True

Comment (IkaDemisto) [reply demisto_sdk/__main__.py]: But it is not internal command.

Comment (bakatzir) [reply demisto_sdk/__main__.py]: I do not think contributors should run this command,
but that we should add it automatically to the their pack_metada.

Comment (IkaDemisto) [reply demisto_sdk/__main__.py]: Hmm it is mainly for contributors, they should define what dependencies are mandatory or not after running this command.
@yuvalbenshalom how do you think we should handle it?

Comment (yuvalbenshalom) [reply demisto_sdk/__main__.py]: @IkaDemisto I think @bakatzir is correct. we need a command that dose "update dependencies", and update the metadata.
we can consider to split it. lets talk about it.

Comment (IkaDemisto) [reply demisto_sdk/__main__.py]: we cannot add this automatically, when you init pack there are no dependencies

Comment (yuvalbenshalom) [reply demisto_sdk/__main__.py]: will create the updating command in a different issue and PR

**Pull request merged**